javaslang/src/main/java/javaslang/control/Try.java

```diff
@@ -55,11 +56,7 @@
      * exception occurs calling {@code supplier.get()}.
      */
     static <T> Try<T> ofSupplier(Supplier<? extends T> supplier) {
-        try {
-            return new Success<>(supplier.get());
-        } catch (Throwable t) {
-            return new Failure<>(t);
-        }
+        return of(supplier::get);
```

Comment:
Yes, we do it everywhere. Seems like ballast but the developer experience (DX) is better. Like a polished  product.